kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/package-info.java

```diff
@@ -38,5 +38,25 @@
  *     <li>{@link io.kroxylicious.proxy.filter.FilterContext#requestFilterResultBuilder()} and</li>
  *     <li>{@link io.kroxylicious.proxy.filter.FilterContext#responseFilterResultBuilder()}.</li>
  * </ul>
+ * <h3 id='implementing.threadSafety'>Thread Safety</h3>
+ * <p>
+ * The Filter API provides the following guarantees:
+ * </p>
+ * <ol>
+ *   <li>There is a single thread associated with each connection.</li>
+ *   <li>The association lasts for the lifetime of connection.</li>
+ *   <li>Each filter instance is associated with exactly one connection.</li>
+ *   <li>Construction of the filter instance and dispatch of the filter methods {@code onXxxRequest} and
+ *       {@code onXxxResponse} are performed by the same connection thread.</li>
+ *   <li>Any computation stages chained to the {@link java.util.concurrent.CompletionStage} returned by
+ *       {@link io.kroxylicious.proxy.filter.FilterContext#sendRequest(short, org.apache.kafka.common.protocol.ApiMessage)}
+ *       using the default execution methods
+ *       (such as {@link java.util.concurrent.CompletionStage#thenApply(java.util.function.Function)})
+ *       or asynchronous default execution methods
+ *       (such as {@link java.util.concurrent.CompletionStage#thenApplyAsync(java.util.function.Function)})
+ *       are guaranteed to be performed by the same connection thread.</li>
```

Comment:
Maybe I need to callout that the guarantee doesn't extend to the java.util.concurrent.CompletionStage#thenApplyAsync(java.util.function.Function, Executor) form? Or is this obvious?

Reply:
I added it.